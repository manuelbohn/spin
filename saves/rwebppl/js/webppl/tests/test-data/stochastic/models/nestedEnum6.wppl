var model = function() {
  var x = uniform(0, 1);

  var outerMarginal = Infer({model() {

    var innerMarginal = Infer({model() {
      var y = flip(x);
      factor(y ? 1 : -1);
      return y;
    }});

    var z = sample(innerMarginal);
    factor(!z ? 0 : -4);
    return z;
  }});

  var w = sample(outerMarginal);
  return x * w;
};
