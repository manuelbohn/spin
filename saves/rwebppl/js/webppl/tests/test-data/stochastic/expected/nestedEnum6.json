{
  "mean": 0.160,
  "std": 0.327
}
